tamr_client/dataset/unified.py

```diff
@@ -0,0 +1,106 @@
+"""
+See https://docs.tamr.com/reference/dataset-models
+"""
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+from tamr_client import response
+from tamr_client.instance import Instance
+from tamr_client.project import Project
+from tamr_client.session import Session
+from tamr_client.types import JsonDict
+from tamr_client.url import URL
+
+
+class NotFound(Exception):
+    """Raised when referencing (e.g. updating or deleting) a unified dataset
+    that does not exist on the server.
+    """
+
+    pass
+
+
+@dataclass(frozen=True)
+class Dataset:
```

Comment:
We've been making class names that are user-facing globally unique to make them easily distinguishable (e.g. if you print the `repr` for an instance of any class). This is a known exception to the "use namespaces instead of redundancy in names" rule.

```suggestion
class UnifiedDataset:
```

If we do this, then we should make sure to hoist the `UnifiedDataset` type so its available at the top-level as `tc.UnifiedDataset`.

We should probably discuss this during OH.

If we continue using this convention, we should probably codify it in the style guide section of the contributor docs.